featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,273 @@
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `entityset`.
+        valid_primitives (List[str]): List of primitives to calculate and check output features.
+    """
+
+    recommended_non_numeric_primitives = set()
+    # Only want to run feature generation on non numeric primitives
+    numeric_columns_to_ignore = list(
+        entityset[target_dataframe_name].ww.select(include="numeric").columns,
+    )
+    features = dfs(
+        entityset=entityset,
+        target_dataframe_name=target_dataframe_name,
+        trans_primitives=valid_primitives,
+        max_depth=1,
+        features_only=True,
+        ignore_columns={target_dataframe_name: numeric_columns_to_ignore},
+    )
+
+    for f in features:
+        if (
+            f.primitive.name is not None
+            and f.primitive.name not in recommended_non_numeric_primitives
+        ):
+            try:
+                matrix = calculate_feature_matrix([f], entityset)
+                for f_name in f.get_feature_names():
+                    if len(matrix[f_name].unique()) > 1:
+                        recommended_non_numeric_primitives.add(f.primitive.name)
+            except Exception:  # If error in calculating feature matrix pass on the recommendation
+                pass
+
+    return recommended_non_numeric_primitives
+
+
+def _recommend_skew_numeric_primitives(
+    numerics_only_df: pd.DataFrame,
+    valid_skew_primtives: set,
+) -> set:
+    """Get a set of recommended skew numeric primitives given an entity set.
+
+    Description:
+        Given woodwork initialized dataframe of origin features with only `numeric` semantic tags and an applicable list of `valid_skew_primitives`,
+        get a set of primitives which could be applied to address right skewness.
+
+    Args:
+        numerics_only_df (pd.DataFrame): Woodwork initialized dataframe of origin features with only `numeric` semantic tags
+        valid_skew_primitives (set): set of valid skew primitives (square_root or natural_logarithm) to potentially recommend.
+
+    Note:
+        We currently only have primitives to address right skewness.
+    """
+    recommended_skew_primitives = set()
+    for col in numerics_only_df:
+        # Shouldn't recommend log, sqrt if nans, zeros and negative numbers are present
+        contains_nan = numerics_only_df[col].isnull().values.any()
+        all_above_zero = (numerics_only_df[col] > 0).all()
+        if all_above_zero and not contains_nan:
+            skew = numerics_only_df[col].skew()
+            # We currently don't have anything in featuretools to automatically handle left skewed data as well as skewed data with negative values
+            if skew > 0.5 and skew < 1 and "square_root" in valid_skew_primtives:
+                recommended_skew_primitives.add("square_root")
+                # TODO: Add Box Cox here when available
```

Comment:
Should we make issues to track these?

Reply:
we have it already : https://github.com/alteryx/featuretools/issues/2070

Reply:
Oh, I meant to to make sure we add it back here!